workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Session/PrepareDeploymentsListState.cs

```diff
@@ -58,9 +58,11 @@ private bool PrepareDeploymentsList()
             var deploymentData = new List<DeploymentData>();
             foreach (var loginToken in Owner.Blackboard.LoginTokens)
             {
-                var data = DeploymentData.CreateFromTags(loginToken.DeploymentName, loginToken.Tags);
-                hasAvailableDeployment |= data.IsAvailable;
-                deploymentData.Add(data);
+                if (DeploymentData.TryToCreateFromTags(loginToken.DeploymentName, loginToken.Tags, out var data))
+                {
+                    hasAvailableDeployment |= data.IsAvailable;
```

Comment:
We can move this out the loop and just do: `deploymentData.Any(data => data.IsAvailable)` ?